### PR TITLE
Initial set up for greenkeeper-lockfile

### PR DESCRIPTION
**Pull request by rsimha-amp**
Based on the steps outlined at https://github.com/greenkeeperio/greenkeeper-lockfile#setup

This PR is a no-op until #12679 is merged. However, since we have tests to ensure that `yarn.lock` is always updated along with `package.json`, this PR must be merged before #12679.

Partial fix for #12181

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -36,6 +39,7 @@ addons:
...
+    secure: "WJxhl/duuCxQW0C5pgpWgrFOvpGabCKhi5IKnyUtaLiWtMXBx4XMYh0ou60zfk5h76cgmXbHcKpHoo6Gyd0enxnwLcnrgP5RdyXPq4lOK8+k/cNNQpXx7VqQl37UohBWeZrEkIFBNOjohlK4jvrOgE4QmZuKwm14dw++CNEI4yNVliLgE6bGyFeaLrVyJufGwDkWKm5aZcq79i+g6BfxYZ5X+otL0XFxaLcWCUUm004taPMYSd1BpZVNuoTnyRsmhSz+wbBdILj7SQMqAj3XIAHAHE4SG9rT7u7Kt236DlAyiIppIL30uttGlS/AHPFpORE6b+kVgO4Xzf4REoGqbK23+TVsdEQJ63NXQcjmFgu9tgNy4+mZ5NDd1rP5zoZTBlo/Co989RgM+gYHJOilc1HYkGOy30O7Im/7FjWPdFmGVmzSYpFPG8zNWM+mNk1Cn74ueyF3GuDw+G5DtVEzwlP8n4qUnVh2TGNkZ8gYlFeUDL0CovvsYPivdK9p2ZQqoLPNKlPC1AqKlo0dyF702NTn36X99q3fMJ/EmspPbaj34Peno1oIAfF4s8zne3l7PiSkAFCFCBUCmyqvwHlvLHJLNtGMf+RG5tlRNZbS7iyW6cQHfftnJrqXuo7cajLU9/5e7qJnyTbu+AUdL+oCYScVU8GQBUHRuPxMHgBZfvs="
```
Comment (rsimha-amp) [inline .travis.yml]: @cramforce This line is a placeholder, and is the result of `travis encrypt GH_TOKEN=foobar`. I'll need to replace it with the encrypted version of the real `GH_TOKEN`, generated as described here: https://github.com/greenkeeperio/greenkeeper-lockfile#setup

File: .travis.yml [hunk 2]
```diff
@@ -35,7 +38,8 @@ addons:
...
+    - secure: "WJxhl/duuCxQW0C5pgpWgrFOvpGabCKhi5IKnyUtaLiWtMXBx4XMYh0ou60zfk5h76cgmXbHcKpHoo6Gyd0enxnwLcnrgP5RdyXPq4lOK8+k/cNNQpXx7VqQl37UohBWeZrEkIFBNOjohlK4jvrOgE4QmZuKwm14dw++CNEI4yNVliLgE6bGyFeaLrVyJufGwDkWKm5aZcq79i+g6BfxYZ5X+otL0XFxaLcWCUUm004taPMYSd1BpZVNuoTnyRsmhSz+wbBdILj7SQMqAj3XIAHAHE4SG9rT7u7Kt236DlAyiIppIL30uttGlS/AHPFpORE6b+kVgO4Xzf4REoGqbK23+TVsdEQJ63NXQcjmFgu9tgNy4+mZ5NDd1rP5zoZTBlo/Co989RgM+gYHJOilc1HYkGOy30O7Im/7FjWPdFmGVmzSYpFPG8zNWM+mNk1Cn74ueyF3GuDw+G5DtVEzwlP8n4qUnVh2TGNkZ8gYlFeUDL0CovvsYPivdK9p2ZQqoLPNKlPC1AqKlo0dyF702NTn36X99q3fMJ/EmspPbaj34Peno1oIAfF4s8zne3l7PiSkAFCFCBUCmyqvwHlvLHJLNtGMf+RG5tlRNZbS7iyW6cQHfftnJrqXuo7cajLU9/5e7qJnyTbu+AUdL+oCYScVU8GQBUHRuPxMHgBZfvs="
```
Comment (rsimha-amp) [inline .travis.yml]: @cramforce This line is a placeholder, and is the result of `travis encrypt GH_TOKEN=foobar`. I'll need to replace it with the encrypted version of the real `GH_TOKEN`, generated as described here: https://github.com/greenkeeperio/greenkeeper-lockfile#setup

Comment (cramforce) [reply .travis.yml]: Do I need to do anything? :)

Comment (rsimha-amp) [reply .travis.yml]: Yes, you need to do the following:
1. Go to https://github.com/settings/tokens and create a GitHub access token with push access to `ampproject/amphtml`
2. Run `travis encrypt GH_TOKEN=<token>` with the token generated in step 1, and send it to me.
3. (Optional) If you aren't set up to run `travis` in step 2, privately share the token from step 1 with me, and I'll run step 2.

Comment (rsimha-amp) [reply .travis.yml]: @cramforce Let me know once you have created the token described above.

Comment (rsimha-amp) [reply .travis.yml]: As discussed with @erwinmombay I've generated a `GH_TOKEN` with just `public_repo` access. Will test and see if that's sufficient.

**Pull request merged**